sql/core/pom.xml

```diff
@@ -86,15 +86,17 @@
       <scope>test</scope>
     </dependency>
 
+    <dependency>
+      <groupId>org.apache.hive</groupId>
+      <artifactId>hive-storage-api</artifactId>
+    </dependency>
```

Comment:
Hi, @wangyum . This PR does lots of things, so it's great.
But, we should not have `hive` dependency in `sql/core` module.
This blocks independent ORC version upgrades. ORC should be upgrade independently like Parquet.
This is the same comment from the original PR. And, it becomes more important because we are using Hive 2 instead of Hive 3.

cc @gatorsmile , @cloud-fan , @rxin

Reply:
Yes. We should not do this. sql/core dos not depend on hive at all.

Reply:
yea, for this one, I wasn't getting it right - we shouldn't do this. Why do we need to do this?

Reply:
The ORC core and mapreduce jars depend on Hive's storage-api: see [ORC-174](https://issues.apache.org/jira/browse/ORC-174).
The Hive's storage-api is a distinct subproject:
https://cwiki.apache.org/confluence/display/Hive/Storage+API+Release+Proposal
https://issues.apache.org/jira/browse/HIVE-15419

Reply:
@wangyum . That's not true. ORC `core/mapreduce` jars (with `nohive` classifier) are designed as a shaded jar to embedded them in different names. As you see, AS-IS `sql/core` works without `hive` dependency at all.